### ISSUE TITLE
DeserializingStream sanity check failed

### ISSUE THREAD

**Issue by kpiyush25** [mit-acl/panther#7]
While running `roslaunch panther simulation.launch quad:=SQ01s`, I am getting the following error and then the process dies (Trying to specify the goal position using `G` or `2D Nav Goal` doesn't work as well) : 
```
[INFO] [1640295936.156948, 0.000000]: Loading model XML from ros parameter robot_description
Starting perfect tracker node for: /SQ01s/
[INFO] [1640295936.208657, 0.000000]: Waiting for service /gazebo/spawn_urdf_model
[INFO] [1640295938.287394, 0.000000]: Loading model XML from file 
/home/piyush/panther_ws/src/panther/panther/meshes/tmp_0.urdf
[INFO] [1640295938.290552, 0.000000]: Waiting for service /gazebo/spawn_urdf_model
[ INFO] [1640295953.298561116]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1640295953.302099473]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, 
waiting...
[ INFO] [1640295955.478527787]: waitForService: Service [/gazebo/set_physics_properties] is now available.
[INFO] [1640295955.502009, 0.000000]: Calling service /gazebo/spawn_urdf_model
[INFO] [1640295955.524182, 0.000000]: Calling service /gazebo/spawn_urdf_model
**DISABLING PHYSICS**
[INFO] [1640295955.601097, 0.001000]: Spawn status: SpawnModel: Successfully spawned entity
[ INFO] [1640295955.607068139, 0.001000000]: Physics dynamic reconfigure ready.
[INFO] [1640295957.281193, 0.197000]: Spawn status: SpawnModel: Successfully spawned entity
[ INFO] [1640295957.297051071, 0.213000000]: Found transform SQ01s --> SQ01s/camera_depth_optical_frame
Basis chosen: MINVO
Parameters obtained, checking them...
par_.fov_depth= 5
par_.Ra= 4.5
par_.drone_radius= 0.2
Parameters checked
Changing DroneStatus from TRAVELING to GOAL_REACHED
[SQ01s/spawn_robot-3] process has finished cleanly
log file: /home/piyush/.ros/log/9f674d0c-6439-11ec-9abd-d0abd521f44e/SQ01s-spawn_robot-3*.log
terminate called after throwing an instance of 'casadi::CasadiException'
  what():  .../casadi/core/serializing_stream.cpp:55: Assertion "check==serialization_check" failed:
DeserializingStream sanity check failed. Expected 123456789012345, but got -5863120225865027243.
[ INFO] [1640295958.262307913, 1.174000000]: Camera Plugin: Using the 'robotNamespace' param: '/SQ01s/'
[ INFO] [1640295958.267886238, 1.180000000]: Camera Plugin (ns = /SQ01s/)  <tf_prefix_>, set to "/SQ01s"
[ WARN] [1640295958.268088816, 1.180000000]: dynamic reconfigure is not enabled for this image topic 
[camera/rgb/image_raw] becuase <cameraName> is not specified
[SQ01s/panther-13] process has died [pid 3137, exit code -6, 
cmd /home/piyush/panther_ws/devel/lib/panther/panther_node ~state:=state ~who_plans:=who_plans 
~term_goal:=term_goal ~trajs_predicted:=tracker_predictor_node/trajs_predicted ~traj:=traj 
~trajs_zhejiang:=trajs_zhejiang ~goal:=goal ~setpoint:=setpoint __name:=panther 
__log:=/home/piyush/.ros/log/9f674d0c-6439-11ec-9abd-d0abd521f44e/SQ01s-panther-13.log].
log file: /home/piyush/.ros/log/9f674d0c-6439-11ec-9abd-d0abd521f44e/SQ01s-panther-13*.log
```
I am using Ubuntu 18.04 and all the packages and dependencies including `casadi` were installed and built properly. Any help in resolving the above issue would be highly appreciated.
Thanks in advance!

**Comment by jtorde** [comment]
Hi @kpiyush25,

I have never found this issue, but some questions to be able to help debug it:

- Does `roscd panther` work? (i.e., does it change your current directory to panther's directory?) Alternatively, try `rospack find panther` in the terminal
- Did you run `git lfs install && git submodule init && git submodule update`? Do you have the .casadi files in the  `panther/matlab/casadi_generated_files` folder?  ([related issue](https://groups.google.com/g/casadi-users/c/FTVvQhlDVVo/m/zeC9tt9pBwAJ))

**Comment by kpiyush25** [comment]
Hi @jtorde 
Thanks for your reply first of all!
I tried all the commands that you asked me to and here's what I got : 

- `roscd panther` works perfectly fine in my system(i.e. takes me to the `panther` directory). 
- `rospack find panther` also works well(i.e. tells the absolute path of the panther package).
- `git lfs install && git submodule init && git submodule update` had also worked fine when I was setting up the package for the first time. I got the following output when I ran the command just now : 
-``` Updated git hooks.
Git LFS initialized. ```
- I have .casadi files in `panther/matlab/casadi_generated_files` and currently I have the following .casadi files in it : ```fit_yaw.casadi                    get_mean_variance_pred_24.casadi  get_mean_variance_pred_39.casadi
get_mean_variance_pred_10.casadi  get_mean_variance_pred_25.casadi  get_mean_variance_pred_40.casadi
get_mean_variance_pred_11.casadi  get_mean_variance_pred_26.casadi  get_mean_variance_pred_4.casadi
get_mean_variance_pred_12.casadi  get_mean_variance_pred_27.casadi  get_mean_variance_pred_5.casadi
get_mean_variance_pred_13.casadi  get_mean_variance_pred_28.casadi  get_mean_variance_pred_6.casadi
get_mean_variance_pred_14.casadi  get_mean_variance_pred_29.casadi  get_mean_variance_pred_7.casadi
get_mean_variance_pred_15.casadi  get_mean_variance_pred_30.casadi  get_mean_variance_pred_8.casadi
get_mean_variance_pred_16.casadi  get_mean_variance_pred_31.casadi  get_mean_variance_pred_9.casadi
get_mean_variance_pred_17.casadi  get_mean_variance_pred_32.casadi  index_instruction.txt
get_mean_variance_pred_18.casadi  get_mean_variance_pred_33.casadi  op.casadi
get_mean_variance_pred_19.casadi  get_mean_variance_pred_34.casadi  op_fixed_pos.casadi
get_mean_variance_pred_20.casadi  get_mean_variance_pred_35.casadi  params_casadi_prediction.yaml
get_mean_variance_pred_21.casadi  get_mean_variance_pred_36.casadi  params_casadi.yaml
get_mean_variance_pred_22.casadi  get_mean_variance_pred_37.casadi  visibility.casadi
get_mean_variance_pred_23.casadi  get_mean_variance_pred_38.casadi```

TL;DR :
Every command that you have suggested above is working fine but the same issue persists while I run the simulation using `roslaunch panther simulation.launch quad:=SQ01s`. 

Waiting for your reply which can help me resolve this issue.
Thanks again!

**Comment by YouShaoze** [comment]
Oh.no...... I have the same problem in my computer (ubuntu18.04 an ROS melodic)

Basis chosen: MINVO
Parameters obtained, checking them...
par_.fov_depth= 5
par_.Ra= 4.5
par_.drone_radius= 0.2
Parameters checked
Changing DroneStatus from TRAVELING to GOAL_REACHED
terminate called after throwing an instance of 'casadi::CasadiException'
  what():  .../casadi/core/serializing_stream.cpp:55: Assertion "check==serialization_check" failed:
DeserializingStream sanity check failed. Expected 123456789012345, but got -5863120225865027243.
[SQ01s/spawn_robot-3] process has finished cleanly
log file: /home/cumt/.ros/log/3fa5a36e-79e7-11ec-aed6-70f11c1522f5/SQ01s-spawn_robot-3*.log
[INFO] [1642679468.823680, 0.000000]: Spawn status: SpawnModel: Successfully spawned entity
[SQ01s/panther-13] process has died [pid 23580, exit code -6, cmd /home/cumt/panther/devel/lib/panther/panther_node ~state:=state ~who_plans:=who_plans ~term_goal:=term_goal ~trajs_predicted:=tracker_predictor_node/trajs_predicted ~traj:=traj ~trajs_zhejiang:=trajs_zhejiang ~goal:=goal ~setpoint:=setpoint __name:=panther __log:=/home/cumt/.ros/log/3fa5a36e-79e7-11ec-aed6-70f11c1522f5/SQ01s-panther-13.log].
log file: /home/cumt/.ros/log/3fa5a36e-79e7-11ec-aed6-70f11c1522f5/SQ01s-panther-13*.log

**Comment by YouShaoze** [comment]
> Oh.no...... I have the same problem in my computer (ubuntu18.04 an ROS melodic)
> 
> Basis chosen: MINVO Parameters obtained, checking them... par_.fov_depth= 5 par_.Ra= 4.5 par_.drone_radius= 0.2 Parameters checked Changing DroneStatus from TRAVELING to GOAL_REACHED terminate called after throwing an instance of 'casadi::CasadiException' what(): .../casadi/core/serializing_stream.cpp:55: Assertion "check==serialization_check" failed: DeserializingStream sanity check failed. Expected 123456789012345, but got -5863120225865027243. [SQ01s/spawn_robot-3] process has finished cleanly log file: /home/cumt/.ros/log/3fa5a36e-79e7-11ec-aed6-70f11c1522f5/SQ01s-spawn_robot-3*.log [INFO] [1642679468.823680, 0.000000]: Spawn status: SpawnModel: Successfully spawned entity [SQ01s/panther-13] process has died [pid 23580, exit code -6, cmd /home/cumt/panther/devel/lib/panther/panther_node ~state:=state ~who_plans:=who_plans ~term_goal:=term_goal ~trajs_predicted:=tracker_predictor_node/trajs_predicted ~traj:=traj ~trajs_zhejiang:=trajs_zhejiang ~goal:=goal ~setpoint:=setpoint __name:=panther __log:=/home/cumt/.ros/log/3fa5a36e-79e7-11ec-aed6-70f11c1522f5/SQ01s-panther-13.log]. log file: /home/cumt/.ros/log/3fa5a36e-79e7-11ec-aed6-70f11c1522f5/SQ01s-panther-13*.log

fit_yaw.casadi                    get_mean_variance_pred_31.casadi
get_mean_variance_pred_10.casadi  get_mean_variance_pred_32.casadi
get_mean_variance_pred_11.casadi  get_mean_variance_pred_33.casadi
get_mean_variance_pred_12.casadi  get_mean_variance_pred_34.casadi
get_mean_variance_pred_13.casadi  get_mean_variance_pred_35.casadi
get_mean_variance_pred_14.casadi  get_mean_variance_pred_36.casadi
get_mean_variance_pred_15.casadi  get_mean_variance_pred_37.casadi
get_mean_variance_pred_16.casadi  get_mean_variance_pred_38.casadi
get_mean_variance_pred_17.casadi  get_mean_variance_pred_39.casadi
get_mean_variance_pred_18.casadi  get_mean_variance_pred_40.casadi
get_mean_variance_pred_19.casadi  get_mean_variance_pred_4.casadi
get_mean_variance_pred_20.casadi  get_mean_variance_pred_5.casadi
get_mean_variance_pred_21.casadi  get_mean_variance_pred_6.casadi
get_mean_variance_pred_22.casadi  get_mean_variance_pred_7.casadi
get_mean_variance_pred_23.casadi  get_mean_variance_pred_8.casadi
get_mean_variance_pred_24.casadi  get_mean_variance_pred_9.casadi
get_mean_variance_pred_25.casadi  index_instruction.txt
get_mean_variance_pred_26.casadi  op.casadi
get_mean_variance_pred_27.casadi  op_fixed_pos.casadi
get_mean_variance_pred_28.casadi  params_casadi_prediction.yaml
get_mean_variance_pred_29.casadi  params_casadi.yaml
get_mean_variance_pred_30.casadi  visibility.casadi

**Comment by YouShaoze** [comment]
I guess there may be a problem with the post installation verification of casadi?

**Comment by jtorde** [comment]
I just did a fresh download/compilation on Ubuntu 20.04 and it's working
So I'm wondering if this has to do with the Ubuntu version used. Could you use the commit d58b025116a97c60fb92535c6c2c0b8b580d43cb instead (and download the submodules accordingly)? (that commit was the last one that was tested on Ubuntu 18.04)

**Comment by ly15029572** [comment]
I have the same problem in my computer (ubuntu20.04 an ROS noetic),but the same issue persists while I run the simulation using roslaunch panther simulation.launch quad:=SQ01s.
my installation process is different from the README in only one place,that is 

linear Sovers   
    cd ~/installations/coinhsl-2015.06.23
but I only can download file is coinhsl-archive-2021.05.05, so I turned this step  into    
   cd ~/installations/coinhsl-archive-2021.05.05

Does this have anything to do with my failure,if so,where should I download coinhsl-2015.06.23.

Waiting for your reply which can help me resolve this issue.
Thanks again!

**Comment by kotakondo** [comment]
I am using docker on Ubuntu20.04/Noetic, and I see the same issue.
